tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_typecast.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-FileCopyrightText: © 2023-24 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+from functools import partial
+import tt_lib as ttl
+
+
+from tests.tt_eager.python_api_testing.sweep_tests import (
+    comparison_funcs,
+    generation_funcs,
+)
+from tests.tt_eager.python_api_testing.sweep_tests.run_pytorch_ci_tests import (
+    run_single_pytorch_test,
+)
+
+mem_configs = [
+    ttl.tensor.MemoryConfig(ttl.tensor.TensorMemoryLayout.INTERLEAVED, ttl.tensor.BufferType.DRAM),
+    ttl.tensor.MemoryConfig(ttl.tensor.TensorMemoryLayout.INTERLEAVED, ttl.tensor.BufferType.L1),
+]
+
+
+@pytest.mark.parametrize(
+    "pt_input_dtype, tt_input_dtype",
+    (
+        (torch.float16, ttl.tensor.DataType.FLOAT32),
+        (torch.float32, ttl.tensor.DataType.BFLOAT8_B),
+        (torch.bfloat16, ttl.tensor.DataType.BFLOAT16),
+        (torch.int, ttl.tensor.DataType.UINT32),
+    ),
+)
+@pytest.mark.parametrize(
+    "pt_output_dtype, tt_output_dtype",
+    (
+        (torch.bfloat16, ttl.tensor.DataType.BFLOAT16),
+        (torch.float32, ttl.tensor.DataType.BFLOAT8_B),
+    ),
+)
+@pytest.mark.parametrize(
+    "input_shapes",
+    [
+        [[1, 1, 32, 32]],  # Single core
+        [[1, 1, 320, 320]],  # multi core
+        [[1, 3, 320, 320]],  # multi core
+        [[1, 1, 32, 32]],  # Single core
+        [[1, 1, 320, 384]],  # Multi core
+        [[1, 3, 320, 384]],  # Multi core
+    ],
+)
+@pytest.mark.parametrize(
+    "input_mem_config",
+    mem_configs,
+)
+@pytest.mark.parametrize(
+    "dst_mem_config",
+    mem_configs,
+)
+class TestTypecast:
+    def test_run_typecast_op(
+        self,
+        pt_output_dtype,
+        tt_output_dtype,
+        pt_input_dtype,
+        tt_input_dtype,
+        input_shapes,
+        input_mem_config,
+        dst_mem_config,
+        device,
+        function_level_defaults,
+    ):
+        if tt_input_dtype in [ttl.tensor.DataType.FLOAT32, ttl.tensor.DataType.UINT32]:
+            pytest.skip(f"{tt_input_dtype} cannot be converted yet. Skip")
+        if tt_input_dtype == tt_output_dtype:
+            pytest.skip("Same I/O data types. Skip.")
```

Comment:
You don't need to skip this since it is supported

Reply:
On device there should be no rounding if your input and output dtypes are the same. It's just copying bytes around.

The only rounding/conversion difference is the initial PT to TT tensor on host.